Un-jank character count indicators

mcginty opened the pull request:
As the final installment of my Winter Cleaning Sunday™

Fixes #1841

moxie0 commented on src/org/thoughtcrime/securesms/TransportOption.java:
```diff
@@ -20,5 +32,8 @@ public boolean isForcedPlaintext() {
-}
 
+  public CharacterState calculateCharacters(int charactersSpent) {
+    return characterCalculator == null ? null : characterCalculator.calculateCharacters(charactersSpent);
```
Instead of returning null and having null checks in the calling code, why not have a push character calculator that always returns 2000 or something?

mcginty replied:
Makes twice as much sense now that we have the max limit! Yep, will do.

mcginty commented:
Added PushCharacterCalculator with 2000 as max size, changed new copyrights to 2015, and renamed `isForcedX` to `isX` since it didn't really make sense.

moxie0 commented:
lgtm
